cli_tools/common/domain/interfaces.go

```diff
@@ -66,9 +65,20 @@ type TarGcsExtractorInterface interface {
 	ExtractTarToGcs(tarGcsPath string, destinationGcsPath string) error
 }
 
-// StorageObjectDeleterInterface represents an object that is responsible for deleting GCS objects
-type StorageObjectDeleterInterface interface {
-	DeleteObject(bucket string, objectPath string) error
+// StorageObjectCreatorInterface represents GCS object creator
+type StorageObjectCreatorInterface interface {
+	GetObject(bucket string, objectPath string) StorageObjectInterface
+}
+
+// StorageObjectInterface represents GCS Object
+type StorageObjectInterface interface {
+	Delete() error
+	GetObjectHandle() *storage.ObjectHandle
+	NewReader() (io.ReadCloser, error)
+	NewWriter() io.WriteCloser
+	ObjectName() string
+	RunComposer(src ...StorageObjectInterface) (*storage.ObjectAttrs, error)
+	RunCopier(src StorageObjectInterface) (*storage.ObjectAttrs, error)
 }
```

Comment:
Any thoughts on making this smaller? For example, an interface for each method?

StorageObjectDeleter
StorageObjectGetter
StorageObjectReader
...

The *huge* benefit is that it clarifies  which dependencies are actually used in a particular context. Also, it avoids the necessity of a big mocking framework, since a particular test can implement a single method on a struct. (I'm not saying that mocking frameworks are bad, but having atomic interfaces gives flexibility)

Reply:
The StorageObjectCreatorInterface is so that we're able to pass in mock objects for testing. I was following an existing pattern for this, but if there is another way to test without having the interface, I'd be happy to remove it. 

I think having the a separate interface for each method is not very user-friendly to the caller. The caller will have to pass in the bucket and object path for each of the call. If caller wants to perform multiple operations on the same object, it can feel redundant.